sagemaker/pipeline/pipeline_parameters.py

```diff
@@ -0,0 +1,764 @@
+"""
+Copyright Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Parameter parsing and validation for GraphStorm SageMaker Pipeline
+"""
+
+import argparse
+import hashlib
+import json
+import logging
+import os
+from dataclasses import asdict, dataclass, fields, is_dataclass
+from typing import List
+
+
+JOB_ORDER = {
+    "gconstruct": 0,
+    "gsprocessing": 1,
+    "dist_part": 2,
+    "gb_convert": 3,
+    "train": 4,
+    "inference": 5,
+}
+
+SCRIPT_PATH = os.path.dirname(os.path.realpath(__file__))
+
+
+@dataclass
+class AWSConfig:
+    """AWS-related configuration"""
+
+    execution_role: str
+    region: str
+    graphstorm_pytorch_cpu_image_url: str
+    graphstorm_pytorch_gpu_image_url: str
+    gsprocessing_pyspark_image_url: str
+
+
+@dataclass
+class InstanceConfig:
+    """Configuration for SageMaker instances"""
+
+    train_infer_instance_count: int
+    cpu_instance_type: str
+    gpu_instance_type: str
+    graph_construction_instance_type: str
+    gsprocessing_instance_count: int
```

Comment:
Do we need the gsprocessing_instance_type?

Reply:
We use `graph_construction_instance_type` for that. Where possible we share `graph_construction*` params between GSProcessing and GConstruct since we only allow one of these two steps to exist in a pipeline.